lib/main.js

```diff
@@ -222,6 +223,17 @@ const Hydrogen = {
       })
     );
 
+    store.subscriptions.add(
+      atom.workspace.addOpener(uri => {
+        if (
+          path.extname(uri).toLowerCase() === ".ipynb" &&
+          atom.config.get("Hydrogen.importNotebookURI") === true
+        ) {
+          return loadNotebook(uri);
+        }
+      })
```

Comment:
What do you think about moving this to `.\import-notebook.js` and importing? That way `main.js` stays cleaner, import notebook logic is isolated ( e.g. `loadNotebook` stays "private" to `importNotebook` to avoid confusion/ambiguity) 

This will make it easier to enhance the opener or import logic down the line if we want and it can be unit tested independently of main.
```js
// import up top becomes:
import { importNotebook, ipynbOpener } from "./import-notebook";
// ...
// then this chunk becomes
store.subscriptions.add(atom.workspace.addOpener(ipynbOpener));
```

Side note: `loadNotebook` is `export`ed only to unit test easier (by skipping over the electron open dialog). We probably should have used the `_` prefix convention and named it `_loadNotebook` initially to indicate that.

Reply:
All good points.